Tests/NestKitTests/FileDownloaderTests.swift

```diff
@@ -0,0 +1,19 @@
+import Foundation
+import Testing
+@testable import NestKit
+
+struct InstallTargetTests {
+    @Test
+    func testNeedsUnzipForZip() throws {
+        let downloader = ZipFileDownloader(urlSession: .shared, fileManager: .default)
+        let url = URL(string: "artifactBundle.zip")!
+        #expect(downloader.needsUnzip(for: url) == true)
```

Comment:
[nit] It might be my preferences.
```suggestion
        #expect(downloader.needsUnzip(for: url))
```